src/Apps/Legion/Services/DataPusherUtils.ts

```diff
@@ -4,176 +4,274 @@ import { ITable } from '../Adapters/Standalone/DataManagement/Models/DataManagem
 import {
     IModelExtended,
     ITwinExtended,
+    PIDSourceUrls,
     PROPERTY_COLUMN_NAME,
+    SourceType,
     TableTypes,
     TIMESTAMP_COLUMN_NAME,
     VALUE_COLUMN_NAME
 } from '../Components/DataPusher/DataPusher.types';
-import { IAppData, IModel, IModelProperty, ITwin } from '../Models/Interfaces';
-import { ICookProperty } from '../Models/Types';
+import {
+    IADXConnection,
+    ICookedSource,
+    IModel,
+    IModelProperty,
+    IPIDDocument,
+    ITwin
+} from '../Models/Interfaces';
+import { ICookProperty, ICookSource } from '../Models/Types';
+import CoffeeRoasteryPIDData from '../Adapters/__mockData__/PID/CoffeeRoastery.json';
+import WasteWaterPIDData from '../Adapters/__mockData__/PID/WasteWater.json';
 
 /**
- * This generates the unique twin to property mapping and then extracting naive model information from those and returning cooked models and twins
- * @param table the source table to cook
- * @param twinIdPropertyColumn the twin id property column
- * @param tableType table schema type, e.g., wide, narrow
- * @returns IAppData which includes models, properties and twins objects
+ * Returns the schema type of a given table
+ * @param table
+ * @returns table schema type (e.g., wide, narrow, tags)
  */
-export const cookSourceTable = (
-    sourceConnectionString: string,
-    table: ITable,
-    twinIdPropertyColumn: string,
-    tableType?: TableTypes
-): IAppData => {
-    const tableSchema =
-        tableType ??
-        (table.Columns.findIndex(
-            (c) => c.columnName === PROPERTY_COLUMN_NAME
-        ) !== -1
-            ? TableTypes.Narrow
-            : table.Columns.findIndex(
-                  (c) => c.columnName === VALUE_COLUMN_NAME
-              ) !== -1
-            ? TableTypes.Tags
-            : TableTypes.Wide);
+export const getTableSchemaTypeFromTable = (table: ITable) => {
+    if (!table) return null;
+    return table.Columns.findIndex(
+        (c) => c.columnName === PROPERTY_COLUMN_NAME
+    ) !== -1
+        ? TableTypes.Narrow
+        : table.Columns.findIndex((c) => c.columnName === VALUE_COLUMN_NAME) !==
+          -1
+        ? TableTypes.Tags
+        : TableTypes.Wide;
+};
 
-    const idxOfTwinIdColumn = table.Columns.findIndex(
-        (c) => c.columnName === twinIdPropertyColumn
-    );
+/**
+ * Based on source type it returns cooked models:
+ * (1) if timeseries type, it generates the unique twin to property mapping and then extracting naive model information from those and returning cooked models and twins
+ * (2) if diagram type, it returns the extracted texts from diagram as twin ids along with other properties like x, y and width, height.
+ * @param sourceType the type of the source
+ * @param source the source to be cooked
+ * @returns ICookedSource which includes models, properties and twins objects
+ */
+export const cookSource = (
+    sourceType: SourceType,
+    source: ICookSource
+): ICookedSource => {
+    switch (sourceType) {
```

Comment:
can we split out the cooking logic for each type to distinct functions? That will make this scale better as we add more types.